src/object/obj_tx.c

```diff
@@ -1917,14 +1960,45 @@ dc_tx_restart(tse_task_t *task)
 			tx->tx_status);
 		rc = -DER_NO_PERM;
 	} else {
+		uint32_t backoff;
+
 		dc_tx_cleanup(tx);
 
-		tx->tx_status = TX_OPEN;
-		tx->tx_epoch.oe_value = 0;
 		if (tx->tx_epoch_task != NULL) {
 			tse_task_decref(tx->tx_epoch_task);
 			tx->tx_epoch_task = NULL;
 		}
+
+		backoff = d_backoff_seq_next(&tx->tx_backoff_seq);
+		if (backoff == 0) {
+			/* No child task is required; reopen the TX now. */
+			dc_tx_restart_reopen(tx);
+		} else {
+			tse_task_t *child;
+
+			/*
+			 * Schedule a delayed child task to implement the
+			 * backoff.
+			 */
+			rc = tse_task_create(dc_tx_restart_reopen_task,
+					     tse_task2sched(task), tx, &child);
+			if (rc != 0)
+				goto out_task;
+			rc = tse_task_register_deps(task, 1, &child);
+			if (rc != 0) {
+				tse_task_decref(task);
+				goto out_task;
```

Comment:
As my understand, delay restart the TX is some kind of optimization to avoid livelock race. If we failed to create the task to delay the restart TX, it is not fatal. It may be better to restart the TX immediately than fail the original TX.

Reply:
Not sure for this, maybe we can retry to commit once. That will give current TX one chance to complete the TX. We assume that the other conflict clients have succeed backoff.

If this TX is still required to be restarted and we still failed to tse_task_register_deps(), then fail out as you current do.

Reply:
Changed to skip the backoff upon these errors.